sale_loyalty_order_line_link/migrations/16.0.1.0.0/post-migration.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def _fill_reward_id_field(env):
+    for line in env["sale.order.line"].filtered("loyalty_program_id"):
+        line.reward_id = env["loyalty.reward"].search(
```

Comment:
@chienandalu 
If the reward_id field does not exist in pre, then neither does the loyalty_reward table.
The SQL query would look like this (it could also be executed in post):
```
UPDATE sale_order_line AS sol
SET reward_id = (
    SELECT lr.id
    FROM loyalty_reward AS lr
    WHERE sol.loyalty_program_id = lr.program_id
)
WHERE sol.loyalty_program_id IS NOT NULL AND sol.loyalty_program_id != '';
```
Unless the base migration respects the ids of the reward_id field in coupon.program (which is not clear to me) and then after creating the field it would be executed:
```
UPDATE sale_order_line AS sol
SET reward_id = (
    SELECT cp.id
    FROM coupon_program AS cp
    WHERE sol.loyalty_program_id = cp.id
)
WHERE sol.loyalty_program_id IS NOT NULL AND sol.loyalty_program_id != '';
```

Reply:
Once you get to your migration script every should be settled down in the base modules, so put yourself in an scenario where everything is properly renamed and transformed. You just have tu pull the loyalty_reward ids that match the loyalty_program ids you already have. Is that simple.

Reply:
OK, done, thanks!